.github/discord-embed-webhook.py

```diff
@@ -13,6 +14,27 @@
 )
 
 
+def get_github_context_vars(context: dict[str, Any]) -> dict[str, str]:
+    context_vars = {}
+    event: dict[str, Any] = context.get("event")
+    if context.get("event_name") == "pull_request":
+        pull_request_context: dict[str, Any] = event.get("pull_request", {})
+        head = pull_request_context.get("head", {})
+        repo = head.get("repo", {})
+        sha = head.get("sha")
+        context_vars["title"] = head.get("ref")
+        context_vars["sha"] = sha
+        context_vars["commit_url"] = f'{repo.get("html_url")}/commit/{sha}'
```

Comment:
I think that `.get()` is overused in many places instead of using `[]`.  In here it's a bug... in other lines maybe less important. If `html_url` would not exist then this would evaluate to sth like `None/commit/324jdsan`
So I think we want explicit failure if it doesn't exist:
```
        context_vars["commit_url"] = f'{repo["html_url"]}/commit/{sha}'
```

but I think that if you expect a certain key to exist

Reply:
hmm, since we're parsing github context data; we should be sure that for specific `event_name`, the payload would be consistent with github's docs https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request